sound/soc/sof/nocodec.c

```diff
@@ -55,35 +38,58 @@ static int nocodec_rtd_init(struct snd_soc_pcm_runtime *rtd)
 	return 0;
 }
 
-/* we just set some BEs - FE provided by topology */
-static struct snd_soc_dai_link sof_nocodec_dais[] = {
-	/* Back End DAI links */
-	{
-		/* SSP0 - Codec */
-		.name = "NoCodec",
-		.id = 0,
-		.init = nocodec_rtd_init,
-		.cpu_dai_name = "sof-nocodec-dai",
-		.platform_name = "sof-audio",
-		.no_pcm = 1,
-		.codec_dai_name = "snd-soc-dummy-dai",
-		.codec_name = "snd-soc-dummy",
-		.ops = &sof_nocodec_ops,
-		.dai_fmt = SND_SOC_DAIFMT_DSP_B | SND_SOC_DAIFMT_NB_NF |
-			SND_SOC_DAIFMT_CBS_CFS,
-		.ignore_suspend = 1,
-		.be_hw_params_fixup = sof_nocodec_codec_fixup,
-		.dpcm_playback = 1,
-		.dpcm_capture = 1,
-	},
-};
-
 static struct snd_soc_card sof_nocodec_card = {
 	.name = "sof-nocodec",
-	.dai_link = sof_nocodec_dais,
-	.num_links = ARRAY_SIZE(sof_nocodec_dais),
 };
 
+int sof_nocodec_setup(struct device *dev,
+		      struct snd_sof_pdata *sof_pdata,
+		      struct snd_soc_acpi_mach *mach,
+		      const struct sof_dev_desc *desc,
+		      struct snd_sof_dsp_ops *ops)
+{
+	struct snd_soc_dai_link *links;
+	char name[32];
+	int i;
+
+	if (!mach)
+		return -EINVAL;
+
+	sof_pdata->drv_name = "sof-nocodec";
+
+	mach->drv_name = "sof-nocodec";
```

Comment:
We are no longer "sof-nocodec" as this code can be used for platforms that support add-in audio cards/devices. We should probably pass this in.

Reply:
Liam's comment was not addressed?

Reply:
I had a call with Liam about this, and aligned to keep this no-codec driver and keep it simple. The reason for it is that we can't leverage deffer probe without using platform probing.